### PR TITLE
fix(query): revoke need consider GrantObject::Database/Table

### LINKED ISSUE #14566: bug: can't revoke privileges from users after upgrade databend-query version to v1.2.321-nightly
### Search before asking

- [X] I had searched in the [issues](https://github.com/datafuselabs/databend/issues) and found no similar issues.

### Version

databend-query: 

select version()
+----------------------------------------------------------------------------------------+
| version()                                                                              |
+----------------------------------------------------------------------------------------+
| 8.0.26-v1.2.322-nightly-d2a0aee647(rust-1.77.0-nightly-2024-01-30T11:35:07.009189580Z) |
+----------------------------------------------------------------------------------------+

databend-meta: 

v1.2.322-nightly

### What's Wrong?

I upgraded databend query from 1.2.241-nightly, and found that I can't revoke the privileges from a user if the privileges are granted with version 1.2.241-nightly. The revoke works if the privileges are granted with version 1.2.321-nightly.

### How to Reproduce?

to reproduce the issue, you need an old version like 1.2.241-nightly databend-query node.

1. in old version databend-query, `grant insert on test_db.test_table to 'test_user';`
2. upgrade the databend-query version to 1.2.321-nightly.
3. in new version databend-query, `revoke insert on test_db.test_table from 'test_user';`
4. in new version databend-query, `show grants for 'test_user';`, you will find the privilege still exists.

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by TCeason**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. revoke need consider GrantObject::Database/Table

2. old version, all contains ownership privilege. now will not revoke ownership privilege.

so the user will contains the object's ownership

Then in bind_revoke, if revoke all on <object> from user, need contains
ownership.

- Fixes #14566

## Tests

- [ ] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14567)

### CODE REVIEW

File: src/query/sql/src/planner/binder/ddl/account.rs [hunk 1]
```diff
@@ -97,8 +98,13 @@ impl Binder {
             AccountMgrSource::ALL { level } => {
                 // ALL PRIVILEGES have different available privileges set on different grant objects
                 // Now in this case all is always true.
-                let grant_object = self.convert_to_grant_object(level).await?;
-                let priv_types = grant_object.available_privileges(false);
+                let grant_object = self.convert_to_revoke_grant_object(level).await?;
+                // Note if old version `grant all on db.*/db.t to user`, the user will contains ownership privilege.
+                // revoke all need to revoke it.
+                let priv_types = match principal {
+                    PrincipalIdentity::User(_) => grant_object[0].available_privileges(true),
```
Comment (TCeason) [inline src/query/sql/src/planner/binder/ddl/account.rs:105]: I think this modify is ok. I test on local with different query version, old query write, new query read:


Prepare data in v1.2.241:

```sql
drop user if exists a;
drop role if exists role1;
drop database if exists a;
create database a;
create user a identified by '123';
grant all on a.* to a;
create role role1;
grant ownership on a.* to role role1;
grant role role1 to a;
```

1.2.241: show grants can not display ownership info. **This is a feature loss in 1.2.241**

```sql
MySQL [(none)]> show grants for a;
+-------------------------------------------------------------+
| Grants                                                      |
+-------------------------------------------------------------+
| GRANT ALL ON 'default'.'a'.* TO 'a'@'%'                     |
| GRANT SELECT ON 'default'.'system'.'one' TO 'a'@'%'         |
| GRANT SELECT ON 'default'.'information_schema'.* TO 'a'@'%' |
+-------------------------------------------------------------+
3 rows in set (0.007 sec)

MySQL [(none)]> show grants for role role1;
+------------------------------------------------------------------+
| Grants                                                           |
+------------------------------------------------------------------+
| GRANT SELECT ON 'default'.'system'.'one' TO ROLE `role1`         |
| GRANT SELECT ON 'default'.'information_schema'.* TO ROLE `role1` |
+------------------------------------------------------------------+
2 rows in set (0.011 sec)
```

====

upgrade to fixed pr version, build databend-query, execute these operator.

user root login:

```sql
MySQL [(none)]> show grants for a;
+-----------------------------------------+
| Grants                                  |
+-----------------------------------------+
| GRANT ALL ON 'default'.'a'.* TO 'a'@'%' |
+-----------------------------------------+
1 row in set (0.008 sec)

MySQL [(none)]> show grants for role role1; -- also can not display ownership info.
Empty set (0.008 sec)


create user b identified by '123' with DEFAULT_ROLE='role1';
grant role role1 to b;

revoke all on a.* from a;
```

user a login: After revoke all privilege from user a, also can access Database a.

```sql
mysql -ua -p123 -h127.0.0.1 -P3307
MySQL [(none)]> show roles;
+--------+-----------------+------------+------------+
| name   | inherited_roles | is_current | is_default |
+--------+-----------------+------------+------------+
| public |               0 |          1 |          0 |
| role1  |               0 |          0 |          0 |
+--------+-----------------+------------+------------+
2 rows in set (0.007 sec)

MySQL [(none)]> set role role1;
Query OK, 0 rows affected (0.006 sec)

MySQL [(none)]> use a
Database changed
MySQL [a]> show tables;
Empty set (0.029 sec)

MySQL [a]> create table t(id int);
Query OK, 0 rows affected (0.068 sec)

MySQL [a]> show tables;
+-------------+
| Tables_in_a |
+-------------+
| t           |
+-------------+
1 row in set (0.039 sec)

MySQL [a]> insert into t values(1);
Query OK, 1 row affected (0.143 sec)
```

user root login : Reovke role1 from a;

```sql                                                                                                                                                                                                                                          
mysql -uroot -h127.0.0.1 -P3307
MySQL [(none)]> show grants for a;
+-------------------------------------------------+
| Grants                                          |
+-------------------------------------------------+
| GRANT OWNERSHIP ON 'default'.'a'.'t' TO 'a'@'%' |
+-------------------------------------------------+
1 row in set (0.013 sec)

MySQL [(none)]> revoke role role1 from a;
Query OK, 0 rows affected (0.047 sec)
```


user b login : Can query Database a.

```sql
mysql -ub -p123 -h127.0.0.1 -P3307

MySQL [(none)]> show roles;
+--------+-----------------+------------+------------+
| name   | inherited_roles | is_current | is_default |
+--------+-----------------+------------+------------+
| public |               0 |          0 |          0 |
| role1  |               0 |          1 |          1 |
+--------+-----------------+------------+------------+
2 rows in set (0.008 sec)

MySQL [(none)]> use a;
Reading table information for completion of table and column names
You can turn off this feature to get a quicker startup with -A

Database changed

MySQL [a]> select * from t;
+------+
| id   |
+------+
|    1 |
+------+
1 row in set (0.040 sec)

```

user a login : Now user a can not query Database a.

```sql
mysql -ua -p123 -h127.0.0.1 -P3307
MySQL [(none)]> use a;
ERROR 1105 (HY000): PermissionDenied. Code: 1063, Text = Permission denied: User 'a'@'%' does not have the required privileges for database 'a'.

MySQL [default]> select * from a.t;
ERROR 1105 (HY000): PermissionDenied. Code: 1063, Text = Permission denied, privilege [Select] is required on 'default'.'a'.'t' for user 'a'@'%' with roles [public].

```

**Comment by TCeason** [comment]
This pr delete some compat version. Please review it @drmingdrmer @dantengsky 

Revoke need respect old version please review it @flaneur2020 @BohuTANG 

Add ci test about compat please review it @everpcpc @drmingdrmer

**Review by flaneur2020 — APPROVED** [review]
LGTM

**Comment by TCeason** [comment]
Try to test system table suit test. Add label ci-benchmark-suites

**Review by drmingdrmer — APPROVED** [review]
> Reviewed 3 of 6 files at r1, 3 of 3 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/datafuselabs/databend/14567)* status: all files reviewed, 1 unresolved discussion (waiting on @BohuTANG, @dantengsky, and @everpcpc)

**Pull request merged**